lib/spack/env/cc

```diff
@@ -709,7 +681,32 @@ categorize_arguments() {
             "$dtags_to_strip")
                 ;;
             *)
-                append return_other_args_list "$1"
+                # if mode is not ld, we can just add to other args
+                if [ "$mode" != "ld" ]; then
+                    append return_other_args_list "$1"
+                    shift
+                    continue
+                fi
+
+                # if we're in linker mode, we need to parse raw RPATH args
+                case "$1" in
+                    -rpath=*)
+                        arg="${1#-rpath=}"
+                        append_path_lists return_rpath_dirs_list "$arg"
+                        ;;
+                    --rpath=*)
+                        arg="${1#--rpath=}"
+                        append_path_lists return_rpath_dirs_list "$arg"
+                        ;;
+                    -rpath|--rpath)
+                        shift
+                        [ $# -eq 0 ] && break  # ignore -rpath without value
```

Comment:
It should still pass this argument to the linker, so that it would error in case a configure script or so adds `-rpath $empty_var`. The same is ensured and I believe tested for -Wl,-rpath.

Reply:
This is addressed. It wasn't tested before but now it's tested for all three cases.